test/com/google/javascript/jscomp/PeepholeSubstituteAlternateSyntaxTest.java

```diff
@@ -335,7 +335,8 @@ public void testUndefined() {
     foldSame("try {} catch(undefined) {}");
     foldSame("for (undefined in {}) {}");
     foldSame("undefined++;");
-    fold("undefined += undefined;", "undefined += void 0;");
+    disableNormalize();
+    foldSame("undefined += undefined;");
```

Comment:
I added a test with `enableNormalize()` to make sure we are folding to `void 0` in that case